cliquet/authentication.py

```diff
@@ -12,37 +12,41 @@
 from cliquet import logger
 
 
-def check_credentials(username, password, request):
+class BasicAuthAuthenticationPolicy(base_auth.BasicAuthAuthenticationPolicy):
     """Basic auth implementation.
 
     Allow any user with any credentials (e.g. there is no need to create an
     account).
 
     """
-    settings = request.registry.settings
-    is_enabled = settings['cliquet.basic_auth_enabled']
-
-    if not is_enabled or not username:
-        return
-
-    hmac_secret = settings['cliquet.userid_hmac_secret'].encode('utf-8')
-    credentials = '%s:%s' % (username, password)
-    userid = hmac.new(hmac_secret,
-                      credentials.encode('utf-8'),
-                      hashlib.sha256).hexdigest()
-
-    # Log authentication context.
-    logger.bind(auth_type='Basic')
-
-    return ["basicauth_%s" % userid]
-
-
-class BasicAuthAuthenticationPolicy(base_auth.BasicAuthAuthenticationPolicy):
     def __init__(self, *args, **kwargs):
-        super(BasicAuthAuthenticationPolicy, self).__init__(check_credentials,
+        noop_check = lambda *a: [Authenticated]  # NOQA
+        super(BasicAuthAuthenticationPolicy, self).__init__(noop_check,
                                                             *args,
                                                             **kwargs)
 
+    def unauthenticated_userid(self, request):
```

Comment:
`unauthenticated_userid` should be called only when we've went trough the authentication procedure already, and it's supposed to returned a cached version of the user id.